src/ServiceWorker.js

```diff
@@ -0,0 +1,63 @@
+/*
+ Based on code under the following license:
+ Copyright 2016 Google Inc. All Rights Reserved.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+     http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+/* eslint-disable no-restricted-globals */
+
+// Names of the two caches used in this version of the service worker.
+// Change to v2, etc. when you update any of the local resources, which will
+// in turn trigger the install event again.
+const PRECACHE = 'v1';
+const RUNTIME = 'runtime';
+
+// A list of local resources we always want to be cached.
+/** @type {string[]} */
+const PRECACHE_URLS = [
+];
+
+// The install handler takes care of precaching the resources we always need.
+self.addEventListener('install', event => {
+    // @ts-ignore
+    event.waitUntil(
+        caches.open(PRECACHE)
+            .then(cache => cache.addAll(PRECACHE_URLS))
+            // @ts-ignore
+            .then(self.skipWaiting()),
+    );
+});
+
+// The activate handler takes care of cleaning up old caches.
+self.addEventListener('activate', event => {
+    const currentCaches = [PRECACHE, RUNTIME];
+    // @ts-ignore
+    event.waitUntil(
+        caches.keys().then(cacheNames => cacheNames.filter(cacheName => currentCaches.indexOf(cacheName) === -1))
+            .then(cachesToDelete => Promise.all(cachesToDelete.map(cacheToDelete => caches.delete(cacheToDelete))))
+            // @ts-ignore
+            .then(() => self.clients.claim()),
+    );
+});
+
+
+// Intercept fetch
+self.addEventListener('fetch', event => {
+    // @ts-ignore
+    if (event.request.url.startsWith(self.location.origin)) {
```

Comment:
From my understanding, whether cookies are send or is not determined by the whole origin.
From https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy:
```
Cookies use a separate definition of origins. A page can set a cookie for its own domain or any parent domain, as long as the parent domain is not a public suffix. Firefox and Chrome use the Public Suffix List to determine if a domain is a public suffix. Internet Explorer uses its own internal method to determine if a domain is a public suffix. The browser will make a cookie available to the given domain including any sub-domains, no matter which protocol (HTTP/HTTPS) or port is used. When you set a cookie, you can limit its availability using the Domain, Path, Secure and Http-Only flags. When you read a cookie, you cannot see from where it was set. Even if you use only secure https connections, any cookie you see may have been set using an insecure connection.
```

Thus, rather the `host` of the location should be checked, ignoring the protocol and port.
It shouldn't make a difference though, as our service worker is only active on `https` and the very only port we're using anyways. It's just a beautification.

Reply:
Well, you just marked this as resolved....
I still think it should be changed.